DNS/Protocol/Domain.cs

```diff
@@ -40,8 +41,15 @@ public class Domain : IComparable<Domain> {
                     ushort pointer = lengthOrPointer.GetBitValueAt(0, 6);
                     offset = (pointer << 8) | message[offset];
 
+                    if (visitedOffsetPointers.Contains(offset)) {
+                        throw new ArgumentException("Compression pointer loop detected");
+                    }
+                    visitedOffsetPointers.Add(offset);
+
                     continue;
-                } else if (lengthOrPointer.GetBitValueAt(6, 2) != 0) {
+                }
+
+                if (lengthOrPointer.GetBitValueAt(6, 2) != 0) {
```

Comment:
Is there a reason for changing this line from an `else if`?

Reply:
This was proposed by "ReSharper", iirc. I think it's because the above `if` anyways uses `continue`, which makes the `else` kind-of unnecessary.